### PR TITLE
Fix minor ledger reported issues

### PR DESCRIPTION
**Pull request by spalmer25**
Ledger report some small issues:
- _They didn't manage to run our tests._
Done in https://github.com/trilitech/ledger-app-tezos-baking/pull/108
- _There are still silenced warnings_
[Makefile: remove silenced warning](https://github.com/trilitech/ledger-app-tezos-baking/commit/7702e732ac52f2fc2c5a66d352ddf92f7b481385)
- _A non properly documented function can be found in `src/to_string.c:344`._
[to_string: improve the copy_string documentation](https://github.com/trilitech/ledger-app-tezos-baking/commit/9f00c8db95540153b6ca5dca3d7f1de64e36c087)
- _A CodeQL identifies `For loop variable changed in the body` in `src/to_string:279`_
[microtez_to_string: clarify the trailing 0s elimination](https://github.com/trilitech/ledger-app-tezos-baking/commit/316208b365752244253a77886fe3ca2de19ac7ba)
- _A CodeQL identifies `Long switch case` in `src/operation.c:355-491`_
Postponed for another MR: https://github.com/trilitech/ledger-app-tezos-baking/issues/109

### CODE REVIEW

**Comment by spalmer25** [comment]
> Can we add a test to check microtez to string conversion with a fractional microtez value with large trailing zeros. A value with no trailing zeros like 19.999999 tez.

Done: [Test: test the tz display](https://github.com/trilitech/ledger-app-tezos-baking/pull/107/commits/ef7e42290c6fbee9fba51e9ab3c367fb7a274ed7)

File: doc/apdu.md [hunk 1]
```diff
@@ -209,6 +209,10 @@ accepted, the signature of the message is returned.
 
 Messages sent in more than one packet will be refused.
 
+If the `message` is a valid `baking message` (`Block` or `Consensus
```
Comment (ajinkyaraj-23) [inline doc/apdu.md:212]: what happens when message is invalid baking message. Does it produce confirmation screen? The language seems to suggest that.

Comment (ajinkyaraj-23) [reply doc/apdu.md:212]:
```suggestion
Signing baking message(block and consensus) does not require user intevention, no confirmation screen will be displayed and the signing process is automatic.
```

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

**Pull request merged**